File: src/examples/InternationalAutocompleteExample.cs
namespace Examples
{
	using System;
    using System.IO;
    using System.Net;
    using SmartyStreets;
	using SmartyStreets.InternationalAutocompleteApi;

    public class InternationalAutocompleteExample
    {
        public static void Run()
		{
            // We recommend storing your secret keys in environment variables.
            var authId = Environment.GetEnvironmentVariable("SMARTY_AUTH_ID");
			var authToken = Environment.GetEnvironmentVariable("SMARTY_AUTH_TOKEN");

			using var client = new ClientBuilder(new BasicAuthCredentials(authId, authToken)).BuildInternationalAutocompleteApiClient();
			
			// Documentation for input fields can be found at:
			// https://smartystreetscom/docs/cloud/international-street-api#http-input-fields

			// Geocoding must be expressly set to get latitude and longitude.
			var lookup = new Lookup("Louis")
			{
				Country = "FRA",
				Locality = "Paris",
			};

			//uncomment the line below to add a custom parameter
			//lookup.AddCustomParameter("max_results", "3");

            try
            { 
	            client.Send(lookup);
            }
            catch (SmartyException ex)
            {
                Console.WriteLine(ex.Message);
                Console.WriteLine(ex.StackTrace);
                return;
            }
            catch (IOException ex)
            {
                Console.WriteLine(ex.StackTrace);
                return;
            }

            var candidates = lookup.Result;
            
            if (candidates == null)
            {
                Console.WriteLine("No candidates. This means the address is not valid.");
                return;
            }
            
            Console.WriteLine();
			Console.WriteLine("*** Results ***");
			foreach (var candidate in candidates)
			{
				if (candidate.AddressText != null) {
					Console.Write(candidate.Entries);
					Console.Write(" ");
					Console.Write(candidate.AddressText);
					Console.Write(", ");
					Console.WriteLine(candidate.AddressID);
				} else {
					Console.Write(candidate.Street);
					Console.Write(" ");
					Console.Write(candidate.Locality);
					Console.Write(", ");
					Console.WriteLine(candidate.CountryISO3);
				}
			}
		}
				
    }
}

File: src/vbexamples/InternationalAutocompleteExample.vb
Imports System.Diagnostics.Metrics
Imports System.Formats
Imports System.IO
Imports SmartyStreets
Imports SmartyStreets.InternationalAutocompleteApi
Module InternationalAutocompleteExample
    Dim authID = Environment.GetEnvironmentVariable("SMARTY_AUTH_ID")
    Dim authToken = Environment.GetEnvironmentVariable("SMARTY_AUTH_TOKEN")
    Dim url = Environment.GetEnvironmentVariable("SMARTY_INTERNATIONAL_AUTOCOMPLETE_URL")

    Dim client = New ClientBuilder(authID, authToken).WithLicense(New List(Of String) From {"international-autocomplete-cloud"}).WithCustomBaseUrl(url).BuildInternationalAutocompleteApiClient()

    Sub InternationalAutocompleteExample()

        Dim lookup = New Lookup("Louis")
        With lookup
            .Country = "FRA"
            .Locality = "Paris"
        End With

        Console.WriteLine("*******************************************************")
        Console.WriteLine()

        Try
            client.Send(lookup)
        Catch ex As SmartyException
            Console.WriteLine(ex.Message)
            Console.WriteLine(ex.StackTrace)
        Catch ex As IOException
            Console.WriteLine(ex.StackTrace)
        Catch ex As Exception
            Console.WriteLine(ex.Message)
            Console.WriteLine(ex.StackTrace)
        End Try

        Dim candidates = lookup.Result

        Console.WriteLine("Original lookup: " + lookup.Search + " " + lookup.Country + Environment.NewLine())

        If candidates Is Nothing Then
            Console.WriteLine("No results. The input is not valid." + Environment.NewLine)
            Return
        End If

        Console.WriteLine("*** Found " + CStr(candidates.Count) + " result" + If(candidates.Count = 1, "", "s") + " ***" + Environment.NewLine)

        For Each candidate In candidates
            Console.WriteLine(candidate.Street + " " + candidate.Locality + ", " + candidate.CountryISO3)
        Next

        Console.WriteLine()

    End Sub

End Module
